File: Python/github_contributions.py
import yaml
import subprocess
from datetime import datetime

# Load existing YAML file
with open('data.yaml', 'r') as file:
  try:
    doc = yaml.safe_load(file)
  except yaml.YAMLError as exc:
    print(exc)

# Check if doc is None
if doc is None:
  doc = {}

# Update the date field
doc['date'] = datetime.now().isoformat()

# Write back to the YAML file
with open('data.yaml', 'w') as file:
  yaml.dump(doc, file)

# Add changes to git
subprocess.run(['git', 'add', '.'])

# Commit changes
subprocess.run(['git', 'commit', '-m', 'Daily commit'])

# Push changes
subprocess.run(['git', 'push'])

print("Daily commit done!")

File: Javascript/github_contributions.js
const fs = require("fs");
const yaml = require("js-yaml");
const { exec } = require("child_process");
// The child_process module in Node.js is used to spawn child processes. These child processes can be used to execute system commands, scripts, or other programs.

// Load existing YAML file
fs.readFile("data.yaml", "utf8", (err, data) => {
  if (err) throw err;

  // Parse YAML file
  const doc = yaml.load(data);

  // Update the date field
  doc.date = new Date().toISOString();

  // Write back to the YAML file
  fs.writeFile("data.yaml", yaml.dump(doc), (err) => {
    if (err) throw err;

    // Add changes to git
    exec("git add .", (err, stdout, stderr) => {
      if (err) throw err;

      // Commit changes
      exec('git commit -m "Daily commit"', (err, stdout, stderr) => {
        if (err) throw err;

        // Push changes
        exec("git push", (err, stdout, stderr) => {
          if (err) throw err;
          console.log("Daily commit done!");
        });
      });
    });
  });
});
